apps/src/code-studio/components/progress/StageLockDialog.jsx

```diff
@@ -0,0 +1,284 @@
+import React from 'react';
+import Radium from 'radium';
+import { connect } from 'react-redux';
+import BaseDialog from '@cdo/apps/templates/BaseDialog';
+import ToggleGroup from '@cdo/apps/templates/ToggleGroup';
+import progressStyles from './progressStyles';
+import { LockStatus, saveLockDialog } from '../../stageLockRedux';
+import color from '../../../color';
+import commonMsg from '@cdo/locale';
+
+const styles = {
+  main: {
+    marginTop: 20,
+    marginBottom: 10,
+    marginLeft: 20,
+    marginRight: 20,
+    color: color.charcoal
+  },
+  title: {
+    color: color.teal,
+    fontSize: 20,
+    fontWeight: 900,
+    marginTop: 15,
+    marginBottom: 15
+  },
+  headerRow: {
+    borderWidth: 1,
+    borderStyle: 'solid',
+    borderColor: color.teal,
+    backgroundColor: color.teal,
+    padding: 10,
+    fontSize: '100%',
+    fontFamily: '"Gotham 4r", sans-serif',
+    fontWeight: 'bold'
+  },
+  descriptionText: {
+    marginTop: 10,
+    marginBottom: 10
+  },
+  studentTable: {
+    width: '100%'
+  },
+  tableCell: {
+    borderWidth: 1,
+    borderStyle: 'solid',
+    borderColor: color.light_gray,
+    padding: 10
+  },
+  selectedCell: {
+    backgroundColor: color.lightest_teal
+  },
+  radioCell: {
+    textAlign: 'center'
+  },
+  buttonContainer: {
+    textAlign: 'right',
+    marginRight: 15
+  }
+};
+
+const StageLockDialog = React.createClass({
+  propTypes: {
+    isOpen: React.PropTypes.bool.isRequired,
+    handleClose: React.PropTypes.func.isRequired,
+    initialLockStatus: React.PropTypes.arrayOf(
+      React.PropTypes.shape({
+        name: React.PropTypes.string.isRequired,
+        lockStatus: React.PropTypes.oneOf(Object.values(LockStatus)).isRequired
+      })
+    ),
+    saving: React.PropTypes.bool.isRequired,
+    saveDialog: React.PropTypes.func.isRequired
+  },
+
+  getInitialState() {
+    return {
+      lockStatus: this.props.initialLockStatus
```

Comment:
[Props in getInitialState is an antipattern](https://facebook.github.io/react/tips/props-in-getInitialState-as-anti-pattern.html).  It looks like you're using state here to manage temporary local state, separate from the save process?  Why can't/shouldn't redux handle all of this?